src/components/site-loading-indicator.tsx

```diff
@@ -0,0 +1,28 @@
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import { useProgressTimer } from '../hooks/use-progress-timer';
+import ProgressBar from './progress-bar';
+
+export function SiteLoadingIndicator( { selectedSiteName }: { selectedSiteName?: string } ) {
+	const { __ } = useI18n();
+
+	const { progress, setProgress } = useProgressTimer( {
+		initialProgress: 5,
+		interval: 1500,
+		maxValue: 95,
+	} );
+
+	useEffect( () => {
+		setProgress( 80 );
+	}, [ setProgress ] );
+
+	return (
+		<div className="flex flex-col w-full h-full app-no-drag-region pt-8 overflow-y-auto justify-center items-center">
+			<div className="w-[300px] text-center">
+				<div className="text-black a8c-subtitle-small mb-4">{ selectedSiteName }</div>
+				<ProgressBar value={ progress } maxValue={ 100 } />
+				<div className="text-a8c-gray-70 a8c-body mt-4">{ __( 'Creating site...' ) }</div>
+			</div>
```

Comment:
I added `Creating site...` here for now as this is for adding a site. I am expecting that the text would be different when the importing functionality is implemented

Reply:
It's good for now. When we work on import, we can refactor to pass the state to display different messages.